### PR TITLE
[FIX] sale_product_pack: fix error when creating Down Payment

### PR DESCRIPTION
**Pull request by augusto-weiss**
Error Traceback:
 `Traceback (most recent call last):
  File "/opt/odoo/odoo/http.py", line 1584, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/opt/odoo/odoo/service/model.py", line 134, in retrying
    result = func()
  File "/opt/odoo/odoo/http.py", line 1611, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/opt/odoo/odoo/http.py", line 1808, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/opt/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/opt/odoo/odoo/http.py", line 699, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/addons/sale/wizard/sale_make_invoice_advance.py", line 149, in create_invoices
    self._create_invoices(self.sale_order_ids)
  File "/opt/odoo/addons/sale/wizard/sale_make_invoice_advance.py", line 177, in _create_invoices
    self.env['sale.order.line'].create(
  File "<decorator-gen-215>", line 2, in create
  File "/opt/odoo/odoo/api.py", line 409, in _model_create_multi
    return create(self, [arg])
  File "/mnt/data/odoo-addons-dir/sale_product_pack/models/sale_order_line.py", line 81, in create
    product = self.env["product.product"].browse(elem["product_id"])
KeyError: 'product_id'`

TO REPLAY:
* Create a sale order and confirm
* From sale order, press Create Invoice
* Select Create Down Payment

image

### CODE REVIEW

**Comment by augusto-weiss** [comment]
Hi @FrancoMaxime !!
Here is an error that we could detect and we added a little fix to avoid this error!

File: sale_product_pack/models/sale_order_line.py [hunk 1]
```diff
@@ -78,8 +78,12 @@ def create(self, vals_list):
         new_vals = []
         res = self.browse()
         for elem in vals_list:
-            product = self.env["product.product"].browse(elem["product_id"])
-            if product.pack_ok and product.pack_type == "detailed":
+            product = (
+                self.env["product.product"].browse(elem["product_id"])
+                if "product_id" in elem
+                else None
+            )
```
Comment (FrancoMaxime) [inline sale_product_pack/models/sale_order_line.py]: I think that could be improved like this, what do you think of this?
```suggestion
            product = self.env["product.product"].browse(elem.get("product_id"))
```

**Review by FrancoMaxime — CHANGES_REQUESTED** [review]

File: sale_product_pack/models/sale_order_line.py (hunk 1 above)
Comment (FrancoMaxime) [reply sale_product_pack/models/sale_order_line.py]: Hi @augusto-weiss, i forgot to mention you during my review. Thanks for the bug fix, maybe you could review the initial pull request, maybe i made others mistakes, i hope not.

**Comment by augusto-weiss** [comment]
Hi @FrancoMaxime  i commited your suggestion!! I'll review your pr!! Thanks

**Comment by FrancoMaxime** [comment]
@augusto-weiss Hi, can you regroup the two commit into the first one? That's better for the OCA repository

**Comment by augusto-weiss** [comment]
Done @FrancoMaxime

**Pull request merged**